### PR TITLE
Create new workflow for xcode-test branch

### PR DESCRIPTION
**Pull request by isabelrios**
This PR would add a new workflow to test and finally enable the xcode-test branch to do its job :) 
This workflow is based in the RunUnitTest workflow which builds the app and runs Unit and SmokeTest. Those checks would be enough to check if the app works well with the newest Xcode versions

### CODE REVIEW

**Comment by isabelrios** [comment]
Fyi, @AaronMT we can sync later to check what steps are needed and/if this looks good to you after latest progress you are Richard made yesterday

**Comment by rpappalax** [comment]
@isabelrios can you please add 3  files to your PR?  
[update.py](https://raw.githubusercontent.com/rpappalax/demo-github-actions/master/update.py)
[requirements.txt](https://raw.githubusercontent.com/rpappalax/demo-github-actions/master/requirements.txt)
[update.py](https://raw.githubusercontent.com/rpappalax/demo-github-actions/master/update.py)

**Comment by AaronMT** [comment]
Pre-script 

```YML
- script@1:
    title: "Check xcode-test branch for recent activity before continuning"
    inputs:
    - content: |-
        #!/usr/bin/env bash
        # fail if any commands fails
        set -e
        # debug log
        set -x 

        YESTERDAY=`date -d "yesterday" '+%Y-%m-%d'`

        pip install jq
        
        curl -s -H 'Accept: application/vnd.github.v3+json' https://api.github.com/repos/mozilla-mobile/firefox-ios/commits\?sha\=xcode-test\&since\=$YESTERDAY | jq -e -r '.[].commit.author | select(.name | contains("BITRISE.YML-bot"))'
        test $? -eq 0 || exit 1
```

**Comment by isabelrios** [comment]
@AaronMT where should the it be placed? between which steps? These are the current ones at the beginning:
Captura de pantalla 2021-01-28 a las 19 22 09

**Comment by AaronMT** [comment]
I believe we intended it to be the first very step. If there's a recent commit in the last day by the bot BITRISE.yml-bot, continue, otherwise exit.

File: bitrise.yml [hunk 1]
```diff
@@ -2794,6 +2794,187 @@ workflows:
...
+            # debug log
+            set -x
+
+            YESTERDAY=`date -d "yesterday" '+%Y-%m-%d'`
+
+            pip install jq
+
+            curl -s -H 'Accept: application/vnd.github.v3+json' https://api.github.com/repos/mozilla-mobile/firefox-ios/commits\?sha\=main\&since\=$YESTERDAY | jq -e -r '.[].commit.author | select(.name | contains("BITRISE.YML-bot"))'
+            test $? -eq 0 || exit 1
+        title: "Check xcode-test branch for recent activity before continuning"
```
Comment (AaronMT) [inline bitrise.yml:2814]: Whoops, I guess this is now main

File: test-fixtures/update.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+'''
+curl ${BITRISE_STACK_INFO} | jq ' . | keys'
+[
+  "available_stacks",
+  "project_types_with_default_stacks",
+  "running_builds_on_private_cloud"
+]
+'''
+pattern = 'osx-xcode-'
+BITRISE_YML = 'bitrise.yml'
```
Comment (AaronMT) [inline test-fixtures/update.py:24]: I think this will need a path adjustment now.

File: bitrise.yml [hunk 3]
```diff
@@ -2794,6 +2794,187 @@ workflows:
...
+    - cache-push@2.4: {}
+    - slack@3.1:
+        inputs:
+        - webhook_url: $WEBHOOK_SLACK_TOKEN
+    description: >-
+      This Workflow is to run tests (currently SmokeTest) when there is a merge
+      in master
+    meta:
+      bitrise.io:
+        stack: osx-xcode-12.0.x
```
Comment (AaronMT) [inline bitrise.yml:2975]: Do we want to initially start on 12.0.x for testing?

File: test-fixtures/update.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    largest_semver = largest_version()
+
+    tmp_file = 'tmp.yml'
+    with open(BITRISE_YML, 'r') as infile:
+        y = yaml.safe_load(infile)
+        current_semver = y['workflows'][WORKFLOW]['meta']['bitrise.io']['stack'] 
+
+        if current_semver == largest_semver:
+            print('XCode version unchanged! aborting.')
```
Comment (AaronMT) [inline test-fixtures/update.py:84]: XCode -> xcode and below

File: bitrise.yml (hunk 3 above)
Comment (isabelrios) [reply bitrise.yml:2975]: I will change to an older one...

File: test-fixtures/update.py (hunk 2 above)
Comment (isabelrios) [reply test-fixtures/update.py]: Changed to `'../bitrise.yml'`

File: bitrise.yml [hunk 5]
```diff
@@ -2794,6 +2794,187 @@ workflows:
         is_expand: false
       BITRISE_SCHEME: Firefox
     description: ''
+  NewXcodeVersions:
```
Comment (isabelrios) [inline bitrise.yml:2797]: Note: This is the first time adding a workflow from here, so let's see how this works...if it appears right away in the Bitrise UI to select it to schedule a build or there is something else on Bitrise side. As a workaround I propose to use for example `aaronmt_chrontab_perftests` modified accordingly and rename it in the future when we know creating/updating workflows from bitrise.yml works.

**Comment by isabelrios** [comment]
@rpappalax @AaronMT let's merge this so that we can monitor how everything works since everything looks fine according to all the tests you have been doing.

**Pull request merged**